modules/db/src/main/Handlers.scala

```diff
@@ -17,22 +17,27 @@ trait Handlers:
 
   // free handlers for all types with TotalWrapper
   // unless they are given an instance of lila.db.NoDbHandler[T]
-  given opaqueHandler[T, A](using
-      sr: SameRuntime[A, T],
+  given opaqueWriter[T, A](using
       rs: SameRuntime[T, A],
-      handler: BSONHandler[A]
-  )(using NotGiven[NoDbHandler[T]]): BSONHandler[T] =
-    handler.as(sr.apply, rs.apply)
+      writer: BSONWriter[A]
+  )(using NotGiven[NoDbHandler[T]]): BSONWriter[T] with
+    def writeTry(t: T) = writer.writeTry(rs(t))
 
-  given listHandler[T: BSONHandler]: BSONHandler[List[T]] with
-    val reader                                 = collectionReader[List, T]
-    val writer                                 = BSONWriter.collectionWriter[T, List[T]]
-    def readTry(bson: BSONValue): Try[List[T]] = reader.readTry(bson)
-    def writeTry(t: List[T]): Try[BSONValue]   = writer.writeTry(t)
+  given opaqueReader[T, A](using
+      sr: SameRuntime[A, T],
+      reader: BSONReader[A]
+  ): BSONReader[T] with
+    def readTry(bson: BSONValue) = reader.readTry(bson).map(sr.apply)
 
-  given userIdOfWriter[U: UserIdOf, T](using writer: BSONWriter[UserId]): BSONWriter[U] with
+  given userIdOfWriter[U: UserIdOf](using writer: BSONWriter[UserId]): BSONWriter[U] with
     inline def writeTry(u: U) = writer.writeTry(u.id)
 
+  given noOpaqueUserId[A: lila.core.userId.OpaqueUserId]: NoDbHandler[A] with {}
+  given noUserIdOf[A: lila.core.userId.UserIdOf]: NoDbHandler[A] with         {}
```

Comment:
This removing `BSONWriter` instance for an
`OpaqueUserId` is used to get `BSONWriter` instance from `UserIdOf` and as newtypes instance, which was fine before and should be fine in 3.6 as well but removing ambiguity is a good thing I guess.